### PR TITLE
Add String methods

### PR DESCRIPTION
**Pull request by petele**

### CODE REVIEW

File: features/strings.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: String
+description: The String object is used to store and manipulate a sequence of characters.
```
Comment (ddbeck) [inline features/strings.yml:2]:
```suggestion
description: The string type (and `String` object) represents a sequence of characters.
```

File: features/strings.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  - javascript.builtins.String.toLowerCase
+  - javascript.builtins.String.toString
+  - javascript.builtins.String.toUpperCase
+  - javascript.builtins.String.toWellFormed
+  - javascript.builtins.String.trim
+  - javascript.builtins.String.trimEnd
+  - javascript.builtins.String.trimStart
+  - javascript.builtins.String.unicode_code_point_escapes
+  - javascript.builtins.String.valueOf
+  - javascript.builtins.String.@@iterator
```
Comment (ddbeck) [inline features/strings.yml:54]:
```suggestion
  - javascript.builtins.String.@@iterator
  - javascript.grammar.string_literals
```

File: features/strings.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
+name: String
+description: The String object is used to store and manipulate a sequence of characters.
+spec: https://tc39.es/ecma262/multipage/text-processing.html#sec-string-objects
```
Comment (ddbeck) [inline features/strings.yml:3]: See https://github.com/web-platform-dx/web-features/pull/1567#pullrequestreview-2232885174. I think we should add a `group` here and include it with other primitive types.

**Review by ddbeck — CHANGES_REQUESTED** [review]

File: features/strings.yml (hunk 3 above)
Comment (petele) [reply features/strings.yml:3]: done

**Review by ddbeck — CHANGES_REQUESTED** [review]
Following up on the `Intl` discussion elsewhere, I'm suggesting dropping the `Intl`-related methods in line comments.

Also: @Elchi3, you and I talked about this (and Number and some others). I think you rightly pointed out that at least some string methods ought to have their own features (e.g., `at()`), for symmetry with the `Array` features. Should we pull that stuff out now? Or would it make more sense to split it up in a separate PR after merging?

File: features/strings.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  - javascript.builtins.String.fromCharCode
+  - javascript.builtins.String.fromCodePoint
+  - javascript.builtins.String.includes
+  - javascript.builtins.String.indexOf
+  - javascript.builtins.String.isWellFormed
+  - javascript.builtins.String.lastIndexOf
+  - javascript.builtins.String.length
+  - javascript.builtins.String.localeCompare
+  - javascript.builtins.String.localeCompare.locales_parameter
+  - javascript.builtins.String.localeCompare.options_parameter
```
[the remark this answers is not in the document]
Comment (Elchi3) [reply features/strings.yml:25]: That's right!

**Comment by Elchi3** [comment]
> Also: @Elchi3, you and I talked about this (and Number and some others). I think you rightly pointed out that at least some string methods ought to have their own features (e.g., `at()`), for symmetry with the `Array` features. Should we pull that stuff out now? Or would it make more sense to split it up in a separate PR after merging?

I think I would pull that out now and possibly figure out in this PR what we want the "String (initial support)" feature to be here (analog to features/array.yml).

**Comment by petele** [comment]
Based on feedback, and to somewhat mirror `Array`, I split string into multiple features, and create a new `String` group to which they all belong.

File: features/string-normalize.yml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+name: String normalize()
+description: The `normalize()` method of strings returns the Unicode Normalization Form of the string as a new string.
```
Comment (ddbeck) [inline features/string-normalize.yml:2]:
```suggestion
description: The `normalize()` method of strings returns a Unicode normal form of a string as a new string. More than one code point sequence can represent the same characters. You can use the `normalize()` method to find canonically or compatibly equivalent strings.
```

File: features/string-pad-startend.yml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+name: String padStart() and padEnd()
```
Comment (ddbeck) [inline features/string-pad-startend.yml:1]: Let's rename this to `string-pad.yml`

File: features/string-pad-startend.yml [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+description: The `padStart()` and `padEnd()` methods of strings return a new string the length specified with padding at the start or end of the string.
```
Comment (ddbeck) [inline features/string-pad-startend.yml:2]:
```suggestion
description: The `padStart()` and `padEnd()` methods of strings return a string lengthened to a minimum number of characters by adding characters to the start or end of the string.
```

File: features/string-raw.yml [hunk 8]
```diff
@@ -0,0 +1,6 @@
+name: String raw()
+description: The `normalize()` static method is a tag function of template literals.
```
Comment (ddbeck) [inline features/string-raw.yml:2]:
```suggestion
description: The `String.raw()` static method interpolates template literal substitutions, but ignores escape sequences. It is the tag function for template literals.
```

Comment (ddbeck) [reply features/string-raw.yml:2]: This one is hard to write about. Florian, if you have a take on this, I'd like to hear it.

File: features/string-repeat.yml [hunk 9]
```diff
@@ -0,0 +1,6 @@
+name: String repeat()
+description: The `repeat()` method of strings returns a new string consisting of the original string repeated `count` times.
```
Comment (ddbeck) [inline features/string-repeat.yml:2]:
```suggestion
description: The `repeat()` method of strings returns the original string repeated a number of times.
```

File: features/string-replaceall.yml [hunk 10]
```diff
@@ -0,0 +1,6 @@
+name: String replaceAll()
+description: The `replaceAll()` method of strings returns a new string where all occurances of `searchFor`, have been replaced with `replaceWith`. `searchFor` may be a string or regular expression.
```
Comment (ddbeck) [inline features/string-replaceall.yml:2]:
```suggestion
description: The `replaceAll()` method of strings returns a new string where all matches of a pattern (a string or regular expression) have been substituted with a replacement string.
```

File: features/string-startsends-with.yml [hunk 11]
```diff
@@ -0,0 +1,7 @@
+name: String startsWith() and endsWith()
+description: The `startsWith()` and `endsWith()` methods of strings returns a boolean indicating if a string start or ends with the provided string.
```
Comment (ddbeck) [inline features/string-startsends-with.yml:2]:
```suggestion
description: The `startsWith()` and `endsWith()` methods of strings returns whether a search string appears at the beginning or end of the provided string.
```

File: features/string-trim-startend.yml [hunk 12]
```diff
@@ -0,0 +1,7 @@
+name: String trimStart() and trimEnd()
+description: The `trimStart()` and `trimEnd()` methods of strings return a new string with whitespace removed from the start or end of the string.
```
Comment (ddbeck) [inline features/string-trim-startend.yml:2]:
```suggestion
description: The `trimStart()` and `trimEnd()` methods of strings return a new string with whitespace removed from the beginning or end of the string.
```

File: features/string-wellformed.yml [hunk 13]
```diff
@@ -0,0 +1,7 @@
+name: String isWellFormed() and toWellFormed
```
Comment (ddbeck) [inline features/string-wellformed.yml:1]:
```suggestion
name: String isWellFormed() and toWellFormed()
```

File: features/string-wellformed.yml [hunk 14]
```diff
@@ -0,0 +1,7 @@
...
+description: The `isWellFormed()` method of strings returns a boolean indicating if the string contains any lone surrogates. `toWellFormed()` returns a new string where all lone surrogates are replaced with the unicode replacement character.
```
Comment (ddbeck) [inline features/string-wellformed.yml:2]:
```suggestion
description: The `isWellFormed()` method of strings returns a boolean indicating if the string contains any Unicode lone surrogates. The `toWellFormed()` method returns a new string where all lone surrogates are replaced by the Unicode replacement character.
```

File: groups/string.yml [hunk 15]
```diff
@@ -0,0 +1,2 @@
+name: String
```
Comment (ddbeck) [inline groups/string.yml:1]: Also rename this file to `strings` (for symmetry with "arrays")

```suggestion
name: Strings
```

**Review by ddbeck — CHANGES_REQUESTED** [review]
I've suggested a number of changes, mostly to names and descriptions. I didn't apply them because a few file renames are needed. Thank you!

Also, @Elchi3, optionally if you want to give this a once-over to compare to the Array features, I'd appreciate it.

**Review by Elchi3 — APPROVED** [review]
LGTM, great work 👍

**Review by ddbeck — APPROVED** [review]

**Pull request merged**